### PR TITLE
[Enhancement] Ventilator Parameters Input - Validate Consultation Bed for Linked Ventilator Asset is resolved

### LINKED ISSUE #7397: [Enhancement] Ventilator Parameters Input - Validate Consultation Bed for Linked Ventilator Asset
## Summary

When users enter Ventilator Parameters for a patient, the system should verify whether the patient's current `consultation_bed` record has a linked Ventilator Asset. If no linked asset is found, the system should prompt the user with a dialog to link a Ventilator being used and capture the time the patient was put on the Ventilator.

## Motivation

This enhancement is crucial for ensuring the integrity of clinical data and for operational efficiency. It prevents the oversight of linking ventilator assets to patients, which is essential for tracking the use of critical equipment and for clinical audits. It also enhances patient safety by ensuring that ventilator use is accurately documented and linked to the patient's record.

## Detailed Description

- When a user saves Ventilator Parameters in the system, initiate a check to confirm if the patient's current `consultation_bed` record has an associated Ventilator Asset.
- If no Ventilator Asset is linked, display a dialog box prompting the user to:
   - Select a Ventilator Asset from a list of available ventilators.
   - Specify the time the patient was put on the ventilator. This time should be logged for accurate medical history and asset tracking.
- Once the user links a Ventilator Asset and specifies the time, the system should create the updated `consultation_bed` record accordingly. 
- Provide appropriate error messages if the linking process fails or if no available Ventilator Assets exist.

## Alternatives Considered

N/A - This specific enhancement addresses a critical workflow gap and improves both clinical data integrity and patient safety. Other alternatives may not provide a direct solution to the issue of unlinked ventilator assets.

## Additional Context

This enhancement is proposed to streamline clinical workflows and ensure that critical equipment usage is accurately recorded against patient records. It not only aids in clinical decision-making but also in inventory management of life-saving equipment.

### PR DESCRIPTION
**Pull request by manasvi-gaur**
- Fixes #7397 
Issue solved

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

File: src/Components/CriticalCareRecording/CriticalCare__API.tsx [hunk 1]
```diff
@@ -24,3 +26,17 @@ export const updateDailyRound = (
...
   });
 };
+
+export const getAsset = (
+  consultationId: string,
+  setAsset: React.Dispatch<React.SetStateAction<number>>
+) => {
+  request(routes.listConsultationBeds, {
+    query: { consultation: consultationId },
+  }).then((result) => {
```
Comment (rithviknishad) [inline src/Components/CriticalCareRecording/CriticalCare__API.tsx]: Pass a `limit` of 1 if only the latest (first) consultation bed record is only required.

Comment (rithviknishad) [inline src/Components/CriticalCareRecording/CriticalCare__API.tsx]:
```suggestion
  }).then(({ data }) => {
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by manasvi-gaur** [comment]
@rithviknishad I have made relevant changes, kindly review the PR

**Review by rithviknishad — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Comment by nihal467** [comment]
image
image

@manasvi-gaur The issue remains unresolved; the validation check isn't functioning properly. I created a ventilator-related parameter input for a patient without a ventilator asset linked to their bed, yet no notification was triggered.

**Comment by manasvi-gaur** [comment]
@nihal467 @rithviknishad, but in my setup its working perfectly without an asset, we can't create a log of critical type with ventilator parameter its showing alert window with functionality to add assets

image

here its notifying to add assets before proceeding while adding ventilator parameter
image

**Comment by rithviknishad** [comment]
@manasvi-gaur I was able to replicate the issue.

If we submit by editing only the following fields shown in the screenshot, we are able to submit the form without linking a ventilator.

It does works only for changes in fields under the Respiratory Support subtitle. 

image


cc: @nihal467

**Comment by manasvi-gaur** [comment]
@nihal467 @rithviknishad done, I have made all relevant changes but for that single condition that was failing, due to my examination I was unable to fix that earlier, sorry for inconvenience

**Comment by nihal467** [comment]
LGTM

**Review by rithviknishad — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**